### DISCUSSION TITLE
What would the recommended approach be to only continue running a task depending on a result of an intermediate step.

### DISCUSSION THREAD

**Discussion by daniel-dsouza** [concourse/concourse#8758] [General]
Hello all,

I currently have a pipeline which builds a project using a Dockerfile with a "dependencies" target and a dependent "build" target. The dependencies target takes a while to assemble but doesn't change frequently, and so I would like to speed up the pipeline by skipping the dependency target build if the configuration (as defined by the output of a shell script) doesn't change between runs.

Ultimately this would break the Concourse idea that a build stage with the same inputs would always have the same output, but is there any way I could implement something like a "no-op" result and have subsequent tasks use the registry artifact from the previous successful run?

My initial thought is to use the try step to run the dependency checking script against the previously built image, and rebuild the dependencies image if necessary.

**Comment by suhlig** [comment]
You could split the docker build step into two jobs and publish the first image to the registry under its own name e.g. `dependencies`.

The second job would `get` the `dependencies` image (via registry resource) and trigger on new versions of it. Docker will find your `dependencies` layer in the registry and will just fetch it (without building it). 

You'd have to version the two layers independently, which seems like the right thing anyway from what you describe.
